### DISCUSSION TITLE
NNN as a buffer list

### DISCUSSION THREAD

**Discussion by ldelossa** [mcchrish/nnn.vim#100] [Ideas]
In my constant quest to remove plugins from my vim setup, I'm curious if there's any feasability of using NNN as a buffer list for Vim. 

This maybe a stretch, but would mean one could use nnn both to replace ctrl-p, nerd tree, and then any buffer list plugins one maybe using.

Thanks.

**Comment by jarun** [comment]
@mizlan please have a look.

**Reply by mizlan** [reply]
This will require major changes to upstream `nnn` because that would mean `nnn` effectively becomes an arbitrary "picker" or "Unix filter" like `fzf`. This would simultaneously mean ignoring most of `nnn`'s niceties and features while adding on a more general picker mode.

I am not onboard with this idea because I feel that it is much too far off of what `nnn` is meant to be. `nnn` is a file manager. Ctrl-P and fzf are general-purpose Unix filters in the manner of fuzzy finders. I would say keep both a filter and a file manager.

N.B.: in the near future you will be able to use `nnn` as a NERDtree replacement, but we are still working on that.

**Reply by jarun** [reply]
Thanks for the notes!

**Reply by ldelossa** [reply]
completely understandable. 

thanks for feedback.

**Reply by N-R-K** [reply]
> that would mean nnn effectively becomes an arbitrary "picker"

That can be done via `nnn -p` . One could create a list of dummy files in a tmpdir (cleanup on exit) and (ab)use the nnn picker mode like that.

---

I don't really use any buffer list plugin, nor am I interested in implementing this. But throwing the idea out for anyone that wants to take a crack.
